tests/toranj/test-039-address-cache-table-snoop.py

```diff
@@ -0,0 +1,418 @@
+    wpan.Node.perform_async_tx_rx()
+    verify(sender.was_successful)
+
+r2_rloc = int(r2.get(wpan.WPAN_THREAD_RLOC16), 16)
+c2_rloc = int(c2.get(wpan.WPAN_THREAD_RLOC16), 16)
+r3_rloc = int(r3.get(wpan.WPAN_THREAD_RLOC16), 16)
+
+# Verify that we do see entries in cache table for all the addresses and all are in "query" state
+
+addr_cache_table = wpan.parse_address_cache_table_result(
+    r1.get(wpan.WPAN_THREAD_ADDRESS_CACHE_TABLE))
+
+verify(len(addr_cache_table) == NUM_QUERY_ADDRS)
+for entry in addr_cache_table:
+    verify(entry.state == wpan.ADDRESS_CACHE_ENTRY_STATE_QUERY)
+    verify(not entry.can_evict())
+    verify(entry.timeout > 0)
+    verify(entry.retry_delay == INITIAL_RETRY_DELAY)
+
+# - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+
+# Check the retry-query behavior
+
+# Wait till all the address queries time out and verify they enter "retry-query" state.
+
+
+def check_cache_entry_switch_to_retry_state():
+    cache_table = wpan.parse_address_cache_table_result(
+        r1.get(wpan.WPAN_THREAD_ADDRESS_CACHE_TABLE))
+    for index in range(NUM_QUERY_ADDRS):
+        verify(cache_table[index].state ==
+               wpan.ADDRESS_CACHE_ENTRY_STATE_RETRY_QUERY)
+        verify(cache_table[index].retry_delay == 2 * INITIAL_RETRY_DELAY)
+
+
+wpan.verify_within(check_cache_entry_switch_to_retry_state, WAIT_TIME)
+
+# Try sending again to same addresses which are in "retry-delay" state.
+
+for num in range(NUM_QUERY_ADDRS):
+    sender = r1.prepare_tx((r1_address, PORT),
+                           (PREFIX + "800:" + str(num), PORT), "hi nobody!", 1)
+    wpan.Node.perform_async_tx_rx()
+    verify(sender.was_successful)
+
+# Make sure the entries stayed in retry-delay as before.
+
+wpan.verify_within(check_cache_entry_switch_to_retry_state, WAIT_TIME)
+
+# Now wait for them to get to zero timeout.
+
+cache_table = wpan.parse_address_cache_table_result(
+    r1.get(wpan.WPAN_THREAD_ADDRESS_CACHE_TABLE))
+
+
+def check_cache_entry_in_retry_state_to_get_to_zero_timeout():
+    cache_table = wpan.parse_address_cache_table_result(
+        r1.get(wpan.WPAN_THREAD_ADDRESS_CACHE_TABLE))
+    for index in range(NUM_ADDRESSES, NUM_QUERY_ADDRS):
+        verify(cache_table[index].state ==
+               wpan.ADDRESS_CACHE_ENTRY_STATE_RETRY_QUERY)
+        verify(cache_table[index].timeout == 0)
+
+
+wpan.verify_within(check_cache_entry_in_retry_state_to_get_to_zero_timeout,
+                   WAIT_TIME)
+
+# Now try again using the same addresses.
+
+for num in range(NUM_QUERY_ADDRS):
+    sender = r1.prepare_tx((r1_address, PORT),
+                           (PREFIX + "800:" + str(num), PORT),
+                           "hi again nobody!", 1)
+    wpan.Node.perform_async_tx_rx()
+    verify(sender.was_successful)
+
+# We expect now after the delay to see retries for same addresses.
+
+
```

Comment:
```suggestion

```

Reply:
I think this one is added by the code style. It wants two lines before a new function `def`.